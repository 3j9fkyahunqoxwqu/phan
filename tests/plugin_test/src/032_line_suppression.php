<?php
/**
 * @property int $prop1
 * @property int $prop2 @phan-suppress-current-line PhanInvalidCommentForDeclarationType
 * @property int $prop3
 * @phan-suppress-next-line PhanInvalidCommentForDeclarationType
 * @property int $prop4
 * @property int $prop5
 */
function test_line_suppression() {
    echo $w; /** @phan-suppress-current-line PhanUndeclaredVariable */
    echo $x; // @phan-suppress-next-line PhanUndeclaredVariable
    echo $y;
    echo $z;

    call_undeclared_function_line_suppressed();  /** @phan-suppress-current-line PhanUndeclaredFunction */
    $result = call_undeclared_function_not_suppressed() + $undefVariable + stdClass::undeclaredMethod();  // @phan-suppress-current-line PhanUndeclaredStaticMethod, PhanUndeclaredVariable description goes here

    // @phan-suppress-current-line PhanUndeclaredVariable should warn about being unused
}
