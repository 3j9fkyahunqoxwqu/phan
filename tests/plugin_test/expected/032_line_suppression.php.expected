src/032_line_suppression.php:3 PhanInvalidCommentForDeclarationType The phpdoc comment for @property cannot occur on a function
src/032_line_suppression.php:5 PhanInvalidCommentForDeclarationType The phpdoc comment for @property cannot occur on a function
src/032_line_suppression.php:8 PhanInvalidCommentForDeclarationType The phpdoc comment for @property cannot occur on a function
src/032_line_suppression.php:10 PhanUnreferencedFunction Possibly zero references to function \test_line_suppression
src/032_line_suppression.php:12 PhanUndeclaredVariable Variable $x is undeclared
src/032_line_suppression.php:14 PhanUndeclaredVariable Variable $z is undeclared
src/032_line_suppression.php:17 PhanUndeclaredFunction Call to undeclared function \call_undeclared_function_not_suppressed()
src/032_line_suppression.php:19 UnusedPluginSuppression Plugin Phan\Plugin\Internal\LineSuppressionPlugin suppresses issue PhanUndeclaredVariable but this suppression is unused or suppressed elsewhere
